compiler-plugin/src/main/kotlin/com/google/devtools/ksp/processing/impl/ResolverImpl.kt

```diff
@@ -290,9 +291,18 @@ class ResolverImpl(
             }
         if (!typeOverride) return false
 
-        return OverridingUtil.DEFAULT.isOverridableBy(
-                superDescriptor, subDescriptor, subClassDescriptor, true
-        ).result == OverridingUtil.OverrideCompatibilityInfo.Result.OVERRIDABLE
+        var queue = listOf(subDescriptor)
+        while (queue.isNotEmpty()) {
+            val newQueue = mutableListOf<CallableMemberDescriptor>()
+            for (overridden in queue) {
+                if (overridden === superDescriptor) {
+                    return true
+                }
+                newQueue.addAll(overridden.overriddenDescriptors.map { it.original })
```

Comment:
I think you can get rid of the above super type check as well.

Also, `recordLookup` may need to be adjusted / inserted accordingly, if it turns out that the original override check is wrong.

Have you tried `OverridingUtil.overrides`? It also has a comment regarding with why it uses structural equivalence rather than ===.